R/html_document.R

```diff
@@ -689,4 +719,49 @@ navbar_link_text <- function(x, ...) {
     tagList(x$text, ...)
 }
 
+add_anchor_sections <- function(anchor_sections, section_divs = FALSE) {
+
+  res <- list(args = NULL, lua_filters = NULL, extra_dependencies = NULL)
 
+  if (xfun::isFALSE(anchor_sections)) return(res)
+
+  if (!pandoc2.0()) {
+    stop("Using anchor_sections requires Pandoc 2.0+", call. = FALSE)
+  }
+
+  # default style used
+  style <- "hash"
+
+    # allowed customization
+  allowed_args <- c("style", "depth")
+  if (is.list(anchor_sections)) {
+    # check list elements
+    all_allowed <- all(names(anchor_sections) %in% allowed_args)
+    if (xfun::isFALSE(all_allowed)) {
+      stop("`anchor_sections` should be TRUE or one of [",
+           paste(allowed_args, collapse = ", "), "]",
+           call. = FALSE)
+    }
+    if ("depth" %in% names(anchor_sections)) {
+      # Add metadata only if required
+      res$args <- pandoc_metadata_arg(
+        "rmd_anchor_depth", anchor_sections[["depth"]]
+      )
+    }
+    # change style if required
+    style <- anchor_sections[["style"]] %||% style
+    if (length(style) != 1) {
+      stop("`anchor_sections` can be a list containing only a single `style` element.",
```

Comment:
Can't you leave argument validation to `html_dependency_anchor_sections()`? 

Currently this error implies that you have a list like `list(style = 1, style = 2)` but I don't think that's actually what you're checking for.

Reply:
I want to check against  this
```r 
expect_error(add_anchor_sections(list(style = c("icon", "dash")), "one of"))
````

My wording is incorrect then.

> Can't you leave argument validation to html_dependency_anchor_sections()?

I put it here at first but thought it was a logic to but outside to keep the dependency function simple as others. 

I'll rethink that.